src/pages/data-management/components/FileUploadModal.jsx

```diff
@@ -0,0 +1,229 @@
+                </Paragraph>
+              </Col>
+              <Col span={24} style={{ textAlign: 'center' }}>
+                <List
+                  dataSource={techOptions[selectedTech].inputInfo}
+                  renderItem={(item) => (
+                    <Paragraph style={{ margin: 0 }}>
+                      &bull;
+                      {` ${item}`}
+                    </Paragraph>
+                  )}
+                />
+              </Col>
+              <Col span={24} style={{ textAlign: 'center', marginTop: '1rem' }}>
+                <Paragraph>{`Drag and drop the folder containing these ${techOptions[selectedTech].inputInfo.length} files. The folder should be named with the sample ID.`}</Paragraph>
+                <Paragraph>
+                  Further guidance on supported file types and formats is available
+                  {' '}
+                  <a rel='noreferrer' target='_blank' href={guidanceFileLink}>here</a>
+                  .
+                </Paragraph>
+              </Col>
+            </Row>
+          ) : ''
+        }
+        <Row>
+          <Col span={24}>
+            <Dropzone onDrop={onDrop}>
+              {({ getRootProps, getInputProps }) => (
+                <div style={{ border: '1px solid #ccc', padding: '2rem 0' }} {...getRootProps({ className: 'dropzone' })} id='dropzone'>
+                  <input {...getInputProps()} />
+                  <Empty description='Drag and drop folders here' image={Empty.PRESENTED_IMAGE_SIMPLE} />
+                </div>
+              )}
+            </Dropzone>
+          </Col>
+        </Row>
+
+        {filesList.length ? (
+          <>
+            <Divider orientation='center'>Uploaded files</Divider>
+            <ul style={{
+              columnCount: 4, listStyleType: 'none', padding: 0, margin: 0,
+            }}
+            >
+              {filesList.map((file, idx) => (
+                <li key={`file-${idx}`}>
+                  <Space>
+                    {file.valid
+                      ? (
+                        <>
+                          <CheckCircleTwoTone twoToneColor='#52c41a' />
+                          {file.name}
+                        </>
+                      ) : (
+                        <>
+                          <CloseCircleTwoTone twoToneColor='#f5222d' />
+                          <span>
+                            {`${file.name} - ${file.errors}`}
+                          </span>
+                        </>
+                      )}
+                    <DeleteOutlined style={{ color: 'crimson' }} onClick={() => { removeFile(idx); }} />
+                  </Space>
+                </li>
+              ))}
+            </ul>
+          </>
+        ) : ''}
+      </Space>
+    </Modal>
+
+  );
+};
+
+NewProjectModal.propTypes = {
+  visible: PropTypes.bool,
+  onUpload: PropTypes.func,
+  onCancel: PropTypes.func,
+};
+
+NewProjectModal.defaultProps = {
+  visible: true,
+  onUpload: null,
```

Comment:
I think I wouldn't consider `onUpload` as a prop that could be null or optional, I would set it as required.

`onCancel` perhaps could be optional but with an empty function as default? Because otherwise we would get an error, right?

Reply:
it's fine to put the `onCancel` to `null` even though we require it as a function in the proptype : https://stackoverflow.com/questions/37842868/how-can-i-declare-a-proptype-corresponding-to-a-nullable-number

Regarding `onUpload`, it's not a prop that's required for rendering the component. That's why I set it to optional, i.e. the component can render even if it doesn't function yet. I think this is a pattern that's also used in UI components library, so we can quickly render the components without having to implement the functionalities yet.